tests/jax/core_system_components/builder_test.py

```diff
@@ -1,106 +1,223 @@
-    (
-        data_server,
-        parameter_server,
-        executor,
-        evaluator,
-        trainer,
-    ) = test_system._builder.store.system_build
+def test_builder() -> Builder:
+    """Dummy builder with no components."""
+    return TestBuilder(
+        components=[],
+        global_config=SimpleNamespace(config_key="config_value"),
+    )
 
-    assert isinstance(parameter_server, ParameterServer)
 
-    assert isinstance(executor, acme.core.Worker)
-    # exec_data_client, env, exec = executor
-    # exec_logger, exec_logger_param = env
-    # assert exec_data_client == data_server
-    # assert exec_logger_param == "param"
-    # assert exec_logger == 1
-
-    # exec_id, exec_adder, exec_param_client, exec_param = exec
-    # assert exec_id == "executor"
-    # assert type(exec_param_client[0]) == ParameterServer
-    # assert exec_param_client[1] == 1
-    # assert exec_adder == 2.7
-    # assert exec_param == 1
-
-    assert isinstance(evaluator, acme.core.Worker)
-    # eval_env, eval_exec = evaluator
-    # eval_id, eval_param_client, eval_exec_param = eval_exec
-    # assert eval_env == env
-    # assert eval_id == "evaluator"
-    # assert eval_param_client == exec_param_client
-    # assert eval_exec_param == exec_param
+def test_global_config_loaded(test_builder: TestBuilder) -> None:
+    """Test that global config is loaded into the store during init()."""
+    assert test_builder.store.global_config.config_key == "config_value"
 
+
+def test_data_server_store(test_builder: TestBuilder) -> None:
+    """Test that store is handled correctly in data_server()."""
+    assert test_builder.data_server() == ["data_table_1", "data_table_2"]
+
+
+def test_parameter_server_store(test_builder: TestBuilder) -> None:
+    """Test that store is handled correctly in parameter_server()."""
+    parameter_server = test_builder.parameter_server()
+    assert isinstance(parameter_server, ParameterServer)
+    assert parameter_server.store == test_builder.store
+
+
+def test_executor_store_when_is_evaluator(test_builder: TestBuilder) -> None:
+    """Test that store is handled correctly in executor() when it's an evaluator."""
+    executor_id = "evaluator"
+    data_server_client = "data_server_client"
+    parameter_server_client = "parameter_server_client"
+    assert (
+        test_builder.executor(
+            executor_id=executor_id,
+            data_server_client=data_server_client,
+            parameter_server_client=parameter_server_client,
+        )
+        == "system_executor"
+    )
+
+    assert test_builder.store.executor_id == executor_id
+    assert test_builder.store.data_server_client == data_server_client
+    assert test_builder.store.parameter_server_client == parameter_server_client
+    assert test_builder.store.is_evaluator
+    assert test_builder.store.adder is None
+
+    assert isinstance(test_builder.store.executor, Executor)
+    assert test_builder.store.executor.store == test_builder.store
```

Comment:
Can we also assert that the correct components were passed in? Same for in other places. Right now it should be an empty list everywhere but perhaps you could just have one component in the builder initially and check that it is passed through properly.

Reply:
Good idea, done!